### ISSUE TITLE
Cannot read property 'onOpen' of null at new AutofocusDirective

### ISSUE THREAD

**Issue by TomThorpe** [dougludlow/ng2-bs3-modal#104]
Hi,

First of all, thanks for writing this.

I'm getting the error `Cannot read property 'onOpen' of null at new AutofocusDirective` when I run my app with ng-bs3-modal added.

It seems like in AutofocusDirective https://github.com/dougludlow/ng2-bs3-modal/blob/master/src/ng2-bs3-modal/directives/autofocus.ts there is the part:

```javascript
if (!modal) {
            this.modal.onOpen.subscribe(() => {
                this.el.nativeElement.focus();
            });
 }
```

But in my case, modal is null (the injection is optional, and I dont have a modal on that page yet). So it crashes on the .onOpen call.

I was wondering whether `if (!modal)` should be `if (modal)`, or whether I'm misunderstanding that part?

Cheers,
Tom

**Comment by dougludlow** [comment]
@TomThorpe, wow, not sure how I missed that one, ha ha. Thanks for letting me know, I'll work on getting a fix out soon as well as some test coverage for that so I don't break it again... 😛

**Comment by TomThorpe** [comment]
No worries :-) I did wonder if I was missing something haha. Easily done though!

Thanks again

**Comment by dougludlow** [comment]
@TomThorpe - sorry it took me so long to get to this, but this is fixed in v0.9.2.

**Comment by TomThorpe** [comment]
Ah great, thanks very much!
